public/app/features/dashboard-scene/sharing/ShareButton/share-externally/ShareExternally.tsx

```diff
@@ -0,0 +1,213 @@
+import ShareAlerts from './ShareAlerts';
+import ShareTypeSelect from './ShareTypeSelect';
+import { isEmailSharingEnabled, PublicDashboardShareType } from './utils';
+
+const selectors = e2eSelectors.pages.ShareDashboardDrawer.ShareExternally;
+
+export const getAnyOneWithTheLinkShareOption = () => {
+  return {
+    label: t('public-dashboard.share-externally.public-share-type-option-label', 'Anyone with the link'),
+    description: t(
+      'public-dashboard.share-externally.public-share-type-option-description',
+      'Anyone with the link can access'
+    ),
+    value: PublicDashboardShareType.PUBLIC,
+    icon: 'globe',
+  };
+};
+
+const getOnlySpecificPeopleShareOption = () => ({
+  label: t('public-dashboard.share-externally.email-share-type-option-label', 'Only specific people'),
+  description: t(
+    'public-dashboard.share-externally.email-share-type-option-description',
+    'Only people with access can open with the link'
+  ),
+  value: PublicDashboardShareType.EMAIL,
+  icon: 'users-alt',
+});
+
+const getShareExternallyOptions = () => {
+  return isEmailSharingEnabled()
+    ? [getOnlySpecificPeopleShareOption(), getAnyOneWithTheLinkShareOption()]
+    : [getAnyOneWithTheLinkShareOption()];
+};
+
+export class ShareExternally extends SceneObjectBase {
+  static Component = ShareExternallyRenderer;
+}
+
+function ShareExternallyRenderer({ model }: SceneComponentProps<ShareExternally>) {
+  const dashboard = getDashboardSceneFor(model);
+  const { data: publicDashboard, isLoading } = useGetPublicDashboardQuery(dashboard.state.uid!);
+  const styles = useStyles2(getStyles);
+
+  if (isLoading) {
+    return <Loader />;
+  }
+
+  return (
+    <div className={styles.container}>
+      <ShareExternallyBase publicDashboard={publicDashboard} />
+    </div>
+  );
+}
+
+function ShareExternallyBase({ publicDashboard }: { publicDashboard?: PublicDashboard }) {
+  const options = getShareExternallyOptions();
+  const getShareType = useMemo(() => {
+    if (publicDashboard && isEmailSharingEnabled()) {
+      const opt = options.find((opt) => opt.value === publicDashboard?.share)!;
+      return opt ?? options[0];
+    }
+
+    return options[0];
+  }, [publicDashboard, options]);
+
+  const [shareType, setShareType] = useState<SelectableValue<PublicDashboardShareType>>(getShareType);
+  const hasWritePermissions = contextSrv.hasPermission(AccessControlAction.DashboardsPublicWrite);
+
+  const Config = useMemo(() => {
+    if (shareType.value === PublicDashboardShareType.EMAIL && isEmailSharingEnabled()) {
+      return <EmailSharing />;
+    }
+
+    return <PublicSharing />;
+  }, [shareType]);
+
+  return (
+    <Stack direction="column" gap={2} data-testid={selectors.container}>
+      <ShareAlerts publicDashboard={publicDashboard} />
+      <ShareTypeSelect setShareType={setShareType} value={shareType} options={options} />
+      {!hasWritePermissions && <NoUpsertPermissionsAlert mode={publicDashboard ? 'edit' : 'create'} />}
```

Comment:
The `NoUpsertPermissionsAlert` warning should be displayed at the top, not in the middle. I couldn't find the design for it.

Reply:
there was a design, so I did it, but I can change it. I think it makes sense to be at the top. I'll add it to the ShareAlerts component